client/src/components/productCarousel.js

```diff
@@ -1,12 +1,28 @@
 import React from 'react';
+import Gallery from './gallery.js';
 
 class ProductCarousel extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-
+      modalSwitch: true
     }
+
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
```

Comment:
when toggling state like this I like to do it with one function `toggleModal() {this.setState({modalSwitch: !this.state.modalSwitch})}`